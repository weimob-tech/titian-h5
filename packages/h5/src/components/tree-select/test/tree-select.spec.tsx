import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { TiTreeSelect } from '../index';
describe('渲染 ti-steps', () => {
  it('基本用法', async () => {
    const options = [
      {
        name: '侧边导航',
        id: 'a1',
        list: [
          { name: '标题文字', id: 'a1-1' },
          { name: '标题文字', id: 'a1-2' },
        ],
      },
    ];
    const alias = { label: 'name', value: 'id', children: 'list' };
    const page = await newSpecPage({
      components: [TiTreeSelect],
      template: () => <ti-tree-select options={options} alias={alias} />,
    });
    expect(page.root).toMatchSnapshot();
  });
});
