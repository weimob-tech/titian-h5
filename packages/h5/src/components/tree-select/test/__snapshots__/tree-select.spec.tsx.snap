// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染 ti-steps 基本用法 1`] = `
<ti-tree-select>
  <mock:shadow-root>
    <div class="titian-tree-select" part="" style="height: 100%;">
      <div class="titian-tree-select-sidebar">
        <ti-sidebar active-index="0">
          <ti-sidebar-item data-index="0" label="侧边导航"></ti-sidebar-item>
        </ti-sidebar>
      </div>
      <div class="titian-tree-select-container">
        <div aria-hidden="true" class="titian-tree-select-cell" data-index="0">
          标题文字
        </div>
        <div aria-hidden="true" class="titian-tree-select-cell" data-index="1">
          标题文字
        </div>
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
</ti-tree-select>
`;
